contracts/contracts/strategies/MorphoCompoundStrategy.sol

```diff
@@ -0,0 +1,260 @@
+        IERC20 rewardToken = IERC20(rewardTokenAddresses[0]);
+        uint256 balance = rewardToken.balanceOf(address(this));
+        emit RewardTokenCollected(
+            harvesterAddress,
+            rewardTokenAddresses[0],
+            balance
+        );
+        rewardToken.safeTransfer(harvesterAddress, balance);
+    }
+
+    /**
+     * @dev Get the amount of rewards pending to be collected from the protocol
+     */
+    function getPendingRewards() external view returns (uint256 balance) {
+        address[] memory poolTokens = new address[](assetsMapped.length);
+        for (uint256 i = 0; i < assetsMapped.length; i++) {
+            poolTokens[i] = assetToPToken[assetsMapped[i]];
+        }
+
+        return ILens(LENS).getUserUnclaimedRewards(poolTokens, address(this));
+    }
+
+    /**
+     * @dev Deposit asset into Morpho
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_asset, _amount);
+    }
+
+    /**
+     * @dev Deposit asset into Morpho
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+
+        IMorpho(MORPHO).supply(
+            address(_getCTokenFor(_asset)),
+            address(this), // the address of the user you want to supply on behalf of
+            _amount
+        );
+        emit Deposit(_asset, address(0), _amount);
+    }
+
+    /**
+     * @dev Deposit the entire balance of any supported asset into Morpho
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        for (uint256 i = 0; i < assetsMapped.length; i++) {
+            uint256 balance = IERC20(assetsMapped[i]).balanceOf(address(this));
+            if (balance > 0) {
+                _deposit(assetsMapped[i], balance);
+            }
+        }
+    }
+
+    /**
+     * @dev Withdraw asset from Morpho
+     * @param _recipient Address to receive withdrawn asset
+     * @param _asset Address of asset to withdraw
+     * @param _amount Amount of asset to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        _withdraw(_recipient, _asset, _amount);
+    }
+
+    function _withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+
+        address pToken = assetToPToken[_asset];
+        uint256 oraclePrice = ORACLE.getUnderlyingPrice(pToken);
+
+        IMorpho(MORPHO).withdraw(pToken, _amount.divPrecisely(oraclePrice));
```

Comment:
I think Morpho's withdraw method works off amounts in the underlying asset? Do we need to divide by the oracle price?

Reply:
This felt a bit unintuitive to me as well. According [to documentation](https://developers.morpho.xyz/get-started/interact-with-morpho/withdraw) after calling withdraw with `CDAI` and `amount` parameters user ends up with an exact `amount` more of DAI (and not cdai).

Reply:
Our withdraw method parameters are also working in the same underlying amounts (DAI, not cDAI). Because both the arguments and morpho are denominated in the same underlying, I don't think we need to make any kind of conversion here? We should just withdraw the raw amount?

Reply:
Yes you are right, I don't know what I was thinking here